includes/class-wp-document-revisions-admin.php

```diff
@@ -1355,8 +1456,34 @@ public function delete_attachments_with_document( $post_id ) {
 
 		$document = get_post( $post_id );
 
-		if ( is_numeric( $document->post_content ) && get_post( $document->post_content ) ) {
-			wp_delete_attachment( $document->post_content, false );
+		// not for an attachment or an autosave revision
+		if ( 'attachment' === $document->post_type || wp_is_post_autosave( $post_id ) ) {
+			return;
+		}
+
+		if ( is_numeric( $document->post_content ) ) {
+			// check that content points to an attachment post
+			$attach = get_post( $document->post_content );
+			if ( isset( $attach->ID ) && 'attachment' === $attach->post_type ) {
+
+				// make sure that the attachment is not refered to by another post
+				global $wpdb;
+				$doc_link = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(1) FROM $wpdb->posts WHERE %d IN (post_parent, id) AND post_content = %d AND post_name NOT LIKE '%d-autosave-v1' ",
```

Comment:
Fixed the PHPCS errors I could. Based on https://wordpress.stackexchange.com/questions/8825/how-do-you-properly-prepare-a-like-sql-statement and https://www.itsupportguides.com/knowledge-base/wordpress/wordpress-how-to-use-like-in-wpdb-prepare-sql-statements/, I think this should be:

```php
$doc_link = $wpdb->get_var(
					$wpdb->prepare(
						"SELECT COUNT(1) FROM $wpdb->posts WHERE %d IN (post_parent, id) AND post_content = %d AND post_name NOT LIKE %s",
						$document->post_parent,
						$document->post_content,
						$wpdb->esc_like( $document->post_parent . '-autosave-v1' )
					)
				);
```

If that still achieves the desired outcome?